src/Microsoft.Health.Fhir.CosmosDb.Core/Configs/CosmosDataStoreConfiguration.cs

```diff
@@ -14,6 +14,16 @@ public class CosmosDataStoreConfiguration
 
         public string Key { get; set; }
 
+        /// <summary>
+        /// When true, the application will use the managed identity of the Azure resource to authenticate to Cosmos DB.
+        /// Key configuration will be ignored if this is set to true.
+        /// </summary>
+        public bool UseManagedIdentity { get; set; }
+
+        public bool AllowDatabaseCreation { get; set; } = true;
+
+        public bool AllowCollectionSetup { get; set; } = true;
```

Comment:
@apurvabhaleMS - We'll need these new settings in managed service :)
```
UseManagedService = true
AllowDatabaseCreation = false
AllowCollectionSetup = false
```

Though in thinking about it, if we add the `Cosmos DB Operator` permission too it would setup the db automatically, like it used to